### PR TITLE
Update build and ui-components

### PR DESCRIPTION
**Pull request by labkey-alan**
#### Rationale
This PR updates the demo module to use the newest version of our build and ui-components

#### Related Pull Requests
* 

#### Changes
* Update build to 2.1.0
* Update ui-components to 2.14.0

### CODE REVIEW

File: demo/package.json [hunk 1]
```diff
@@ -31,6 +31,9 @@
   },
   "jest": {
     "globals": {
+      "ts-jest": {
+        "tsconfig": "node_modules/@labkey/build/webpack/tsconfig.json"
```
Comment (cnathe) [inline demo/package.json:35]: any reason that the demo module tsconfig.json file wasn't removed in this module?

**Review by cnathe — CHANGES_REQUESTED** [review]
also, to get the scss paths to work for the @labkey/components assets in both dev/prod and watch mode, I had to introduce aliases for them as well. I believe you'll have to update that line in the queryModelExample.scss file:
from `@import "~@labkey/components/dist/assets/scss/theme";` to `@import "@labkey/components-scss";`

finally, was the main.d.ts file for the LABKEY var not needed in this module? It looks like the QueryModelPage entryPoint uses @labkey/components so I would expect to see those typings errors on start-link

**Comment by cnathe** [comment]
> also, to get the scss paths to work for the @labkey/components assets in both dev/prod and watch mode, I had to introduce aliases for them as well. I believe you'll have to update that line in the queryModelExample.scss file:
> from `@import "~@labkey/components/dist/assets/scss/theme";` to `@import "@labkey/components-scss";`
> 
> finally, was the main.d.ts file for the LABKEY var not needed in this module? It looks like the QueryModelPage entryPoint uses @labkey/components so I would expect to see those typings errors on start-link

Ah, I just noticed that the demo/package.json doesn't even have the `start-link` script command. I think we should add it.

**Comment by labkey-alan** [comment]
I don't think we should add a start-link command to this project. This is not a project that we are often building against when actively developing on our components repo.

**Review by cnathe — APPROVED** [review]
I get your point about why we wouldn't add `start-link` here. Can we remove the tsconfig.json file from this module?

**Pull request merged**